_hooks/pre-commit

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+
+# A git hook script that checks for trailing whitespace
+# in your commits. Bypass it with the --no-verify option
+# to git-commit
+#
+# usage: make a soft link to this file,
+# e.g., ln -s ~/config/pre-commit.git.sh ~/some_project/.git/hooks/pre-commit
+
+if test "$(git diff --check --cached)"
```

Comment:
OK, I have incorporated the travis checks into this hook now as well as rebased from master to pull in the latest changes.  In redoing the hook code to pull in all of the travis checks there are a couple of consequences:
1. I didn't like hardcoding the same travis commands in the pre-commit hook AND the .travis file, so I created a new file called _tests/travis-checks.sh that can then be run in the hook & on travis and only has to be maintained in one spot.  Since the hook is not to check external links, but it should on travis, the hook will pass in an argument to disable external link checks.
2. In order to do all the travis checks in the hook, the sed config alteration also needed to be done.  I didn't like how it was altering the _config.yml file in place, so I think it would be better to now alter it to a _config-deploy.yml file so we aren't changing files under source control and noone has a chance of accidentally commiting it.  I added the _config-deploy.yml to the .gitignore to prevent any commit issues with that file too.
3. Since we took out the ws check, we are relying on the markdown linter for whitespace checking.  I didn't add any instructions for fixing ws issues since it could really fail for any number of reasons.  If we want to grep on the issues and add instructions to fixing whitespace issues ONLY IF found in the grep, we can do that, but adds more complexity to the bash script which I think we were trying to avoid.